src/teraranger_evo.cpp

```diff
@@ -0,0 +1,215 @@
+#include <ros/console.h>
+#include <string>
+
+#include <teraranger_hub/RangeArray.h>
+#include <teraranger_hub/teraranger_evo.h>
+#include <teraranger_hub/helper_lib.h>
+
+namespace teraranger_hub
+{
+
+TerarangerHubEvo::TerarangerHubEvo()
+{
+    // Get parameters and namespace
+    ros::NodeHandle private_node_handle_("~");
+    private_node_handle_.param("portname", portname_,
+                               std::string("/dev/ttyACM0"));
+    ns_ = ros::this_node::getNamespace();
+    ns_ = ros::names::clean(ns_);
+    ROS_INFO("node namespace: [%s]", ns_.c_str());
+
+    // Publishers
+    range_publisher_ = nh_.advertise<teraranger_hub::RangeArray>("teraranger_evo", 8);
+
+    // Serial Port init
+    serial_port_ = new SerialPort();
+    if (!serial_port_->connect(portname_)) {
+      ROS_ERROR("Could not open : %s ", portname_.c_str());
+      ros::shutdown();
+      return;
+    }
+    else {
+      serial_data_callback_function_ =
+          boost::bind(&TerarangerHubEvo::serialDataCallback, this, _1);
+      serial_port_->setSerialCallbackFunction(&serial_data_callback_function_);
+    }
+
+    // Output loaded parameters to console for double checking
+    ROS_INFO("[%s] is up and running with the following parameters:",
+             ros::this_node::getName().c_str());
+    ROS_INFO("[%s] portname: %s", ros::this_node::getName().c_str(),
+             portname_.c_str());
+
+    //Initialize local parameters and measurement array
+    field_of_view = 0.0593;
+    max_range = 14.0;
```

Comment:
Just checked the fov should be 2 degrees (remember that the value here is in radians though)